GP/gp.c

```diff
@@ -1648,7 +1648,8 @@ GPResult gpGetBuddyStatus(
 	/////////////////////////////
 	if(iconnection->simulation)
 	{
-		memset(status, 0, sizeof(GPBuddyStatus));
+		if(status != NULL)
```

Comment:
While there might a null pointer error without this guard, the next line specifies that status must not be null.

I would remove this check and just move 
	// Check for a NULL status.
	///////////////////////////
	if(status == NULL)
		Error(connection, GP_PARAMETER_ERROR, "Invalid status.");
before this simulation block

Reply:
Not sure with this, as looks like in simulation mode we never set error and always return `GP_NO_ERROR`.

`Error` in such case replaced with `{gpiSetErrorString(connection, string); return (result);}`

Reply:
Hmm you're right here, the simulation mode should not return any error (well it should not even used to begin with)